automation/Extension/firefox/data/content.js

```diff
@@ -351,106 +354,180 @@ function getPageScript() {
-    function instrumentObject(object, objectName, prototype=false, logSettings={}) {
+    function isObject(object, propertyName) {
+      try {
+        var property = object[propertyName];
+      } catch(error) {
+        return false;
+      }
+      if (property === null) { // null is type "object"
+        return false;
+      }
+      return typeof property == 'object';
+    }
+
+    function instrumentObject(object, objectName, logSettings={}) {
+      // Use for objects or object prototypes
+      //
+      // Parameters
+      // ----------
+      //   object : Object
+      //     Object to instrument
+      //   objectName : String
+      //     Name of the object to be instrumented (saved to database)
+      //   logSettings : Object
+      //     (optional) object that can be used to specify additional logging
+      //     configurations. See available options below.
+      //
+      // logSettings options (all optional)
+      // -------------------
+      //   propertiesToInstrument : Array
+      //     An array of properties to instrument on this object. Default is
+      //     all properties.
+      //   excludedProperties : Array
+      //     Properties excluded from instrumentation. Default is an empty
+      //     array.
+      //   logCallStack : boolean
+      //     Set to true save the call stack info with each property call.
+      //     Default is `false`.
+      //   logFunctionsAsStrings : boolean
+      //     Set to true to save functional arguments as strings during
+      //     argument serialization. Default is `false`.
+      //   recursive : boolean
+      //     Set to `true` to recursively instrument all object properties of
+      //     the given `object`. Default is `false`
+      //     NOTE:
+      //       (1)`logSettings['propertiesToInstrument']` does not propagate
+      //           to sub-objects.
+      //       (2) Sub-objects of prototypes can not be instrumented
+      //           recursively as these properties can not be accessed
+      //           until an instance of the prototype is created.
+      //   depth : integer
+      //     Recursion limit when instrumenting object recursively.
+      //     Default is `5`.
       var properties = logSettings.propertiesToInstrument ?
         logSettings.propertiesToInstrument : Object.getPropertyNames(object);
       for (var i = 0; i < properties.length; i++) {
         if (logSettings.excludedProperties &&
             logSettings.excludedProperties.indexOf(properties[i]) > -1) {
           continue;
         }
-        if (prototype) {
-          instrumentPrototypeProperty(object, objectName, properties[i], logSettings);
-        } else {
-          instrumentObjectProperty(object, objectName, properties[i], logSettings);
-        }
-      }
-    }
+        if (!!logSettings.recursive && properties[i] != '__proto__' &&
+            isObject(object, properties[i]) &&
+            (!('depth' in logSettings) || logSettings.depth > 0)) {
```

Comment:
This beefy if condition was a bit difficult for me to follow, esp. the last line :)

I understand that this is the stopping condition for the recursion, so maybe include a short comment to explain what is being checked here.